include/swift/AST/DiagnosticsSema.def

```diff
@@ -1908,6 +1908,10 @@ ERROR(inheritance_from_final_class,none,
 ERROR(inheritance_from_unspecialized_objc_generic_class,none,
       "inheritance from a generic Objective-C class %0 must bind "
       "type parameters of %0 to specific concrete types", (Identifier))
+ERROR(inheritance_from_class_with_missing_vtable_entries,none,
+      "cannot inherit from class %0 in Swift %1 mode because it has "
+      "overridable members that cannot be loaded",
```

Comment:
Since version mix and match is the most likely cause of deserialization recovery for Swift 4, maybe having a specialized diagnostic for that case is good enough.